src/WooCommerce.php

```diff
@@ -186,6 +186,29 @@ public static function woocommerce_get_availability($stock, $product) {
     return $stock;
   }
 
+  /**
+   * Cron event callback to remove outdated back-in-stock product metadata.
+   */
+  public static function cron_ensure_back_in_stock() {
+    global $wpdb;
+    $ids = $wpdb->get_col($wpdb->prepare("SELECT ID FROM {$wpdb->posts}
+      INNER JOIN {$wpdb->postmeta} pm ON {$wpdb->posts}.ID = pm.post_id
+      LEFT JOIN {$wpdb->postmeta} moeve ON {$wpdb->posts}.ID = moeve.post_id AND moeve.meta_key LIKE %s
+      WHERE pm.meta_key = %s
+        AND pm.meta_value <= %s
+        AND moeve.meta_id IS NULL", [
+          '_woocommerce-moeve_id_%',
+          '_' . Plugin::PREFIX . '_back_in_stock_date',
+          date('Y-m-d H:i:s'),
```

Comment:
"Any time today" should be considered obsolete, so we should omit the time.
```suggestion
          date('Y-m-d'),
```